tls/s2n_connection.h

```diff
@@ -36,6 +36,22 @@
 
 #define S2N_TLS_PROTOCOL_VERSION_LEN    2
 
+typedef enum { S2N_CERT_AUTH_REQUIRED, S2N_CERT_AUTH_NONE } s2n_cert_auth_type;
```

Comment:
Perhaps this is a personal preference.   I would reverse the order of enum vals, so that the default (S2N_CERT_AUTH_NONE) comes to 0.